specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json

```diff
@@ -0,0 +1,1317 @@
+        }
+      },
+      "description": "Configuration setting of LCM (Local Configuration Manager)."
+    },
+    "AssignmentReport": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "ARM resource id of the report for the guest configuration assignment."
+        },
+        "reportId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "GUID that identifies the guest configuration assignment report under a subscription, resource group."
+        },
+        "assignment": {
+          "$ref": "#/definitions/AssignmentInfo",
+          "description": "Configuration details of the guest configuration assignment."
+        },
+        "vm": {
+          "$ref": "#/definitions/VMInfo",
+          "description": "Information about the VM."
+        },
+        "startTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "Start date and time of the guest configuration assignment compliance status check."
+        },
+        "endTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "End date and time of the guest configuration assignment compliance status check."
+        },
+        "complianceStatus": {
+          "type": "string",
+          "readOnly": true,
+          "description": "A value indicating compliance status of the machine for the assigned guest configuration.",
+          "x-ms-enum": {
+            "name": "complianceStatus",
+            "modelAsString": true
+          },
+          "enum": [
+            "Compliant",
+            "NonCompliant",
+            "Pending"
+          ]
+        },
+        "operationType": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Type of report, Consistency or Initial",
+          "x-ms-enum": {
+            "name": "type",
+            "modelAsString": true
+          },
+          "enum": [
+            "Consistency",
+            "Initial"
+          ]
+        },
+        "resources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssignmentReportResourceMetadata"
+          },
+          "description": "The list of resources for which guest configuration assignment compliance is checked."
+        }
+      }
+    },
+    "GuestConfigurationAssignmentProperties": {
+      "properties": {
+        "targetResourceId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "VM resource Id.",
+          "x-nullable": true
+        },
+        "subscriptionId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Subscription Id to which the Guest Configuration assignment resource is assigned to.",
+          "x-nullable": true
+        },
```

Comment:
these both come from the resource ID and should not be duplicated within the resource

Reply:
these new properties are needed by the ARG team. I can remove the subscriptionId bt targetResourceId is being used for queries.

Reply:
for the purpose of establishing the correlation id's - For cross dataset references/ joins. since guest config reosurce is a proxy resource and the other data set is a tracked - arg team cant do queries based on the default resource ID. so they asked us to add this property so they can use this for joins and stuff